lib/system/iterators_1.nim

```diff
@@ -38,6 +37,7 @@ iterator countdown*[T](a, b: T, step: Positive = 1): T {.inline.} =
     var res = a
     while res >= b:
       yield res
+      if res <= succ(b, step.Natural - 1): break
```

Comment:
Not acceptable. Bloat plus a false sense of "correctness". If you bump into the boundaries of the numeric types that you use, use different numeric types such as a BigInt.

Reply:
Any unsigned type stops at 0, regardless of how wide it is. Things like `countdown(5.uint64, 0.uint64, 2)` won't work.